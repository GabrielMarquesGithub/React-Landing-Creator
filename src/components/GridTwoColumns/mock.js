export default {
  title: 'Grid two columns',
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Animi, illo minima. Nesciunt debitis et accusamus cupiditate, ipsam eaque, excepturi nobis deserunt inventore expedita quod beatae, voluptatibus eius quos autem natus.',
  image: 'assets/images/bgImg.svg',
  background: true,
};
