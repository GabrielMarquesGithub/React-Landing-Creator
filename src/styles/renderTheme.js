import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export const renderTheme = (element) => {
  return render(<ThemeProvider theme={theme}>{element}</ThemeProvider>);
};
