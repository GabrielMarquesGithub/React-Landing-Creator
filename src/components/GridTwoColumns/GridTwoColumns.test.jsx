import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { GridTwoColumns as GridTwoColumns } from '.';

import mock from './mock';

describe('<GridTwoColumns />', () => {
  it('should render correct', () => {
    const { container } = renderTheme(<GridTwoColumns {...mock} />);
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
