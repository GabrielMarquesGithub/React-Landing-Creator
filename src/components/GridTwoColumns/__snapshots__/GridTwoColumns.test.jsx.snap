// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumns /> should render correct 1`] = `
<div>
  <div
    class="sc-dkzDqf jqUFFT"
  >
    <section
      class="sc-gsnTZi jiuFls"
    >
      <div
        class="sc-eCYdqJ ckbRfo"
      >
        <div
          class="sc-jSMfEi eRVEku"
        >
          <h2
            class="sc-bczRLJ esykyK"
          >
            Grid two columns
          </h2>
          <p
            class="sc-hKMtZM SUpSD"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Animi, illo minima. Nesciunt debitis et accusamus cupiditate, ipsam eaque, excepturi nobis deserunt inventore expedita quod beatae, voluptatibus eius quos autem natus.
          </p>
        </div>
        <div
          class="sc-gKXOVf dgdQVN"
        >
          <img
            alt="Grid two columns"
            src="assets/images/bgImg.svg"
          />
        </div>
      </div>
    </section>
  </div>
</div>
`;
